packages/design-system/src/scalars/components/number-field/number-field.tsx

```diff
@@ -185,6 +185,22 @@ export const NumberFieldRaw = forwardRef<HTMLInputElement, NumberFieldProps>(
       onChange?.(nativeEvent as unknown as React.ChangeEvent<HTMLInputElement>);
       onBlur?.(e);
     };
+
+    // Avoid to write letters directly in safari
+    const handleInput = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      const char = event.key;
+      // Allow numbers, decimal point, hyphen (for negatives) and control keys
+      if (
+        !/^[0-9.-]$/.test(char) && // Numbers, decimal point and hyphen
+        !["Backspace", "Delete", "ArrowLeft", "ArrowRight", "Tab"].includes(
```

Comment:
Why do we need to prevent those keys, they should be necessary for a good UX

Reply:
Its need for the  safari, but i will check only for the letter, done